### PR TITLE
DEV-76: Parallel deploy

### PR DESCRIPTION
**Pull request by uptickmetachu**
Parallelise helm installs.

Changes:
- Runs helm update/helm uninstall in parallel
- Changes the `run` function to use asyncio.subprocess, a native async shell runner.
- Renames the `run` argument `catch -> suppress_errors` be more precise. 
- Passes around TypedDicts for better type error checking.

### CODE REVIEW

File: gitops_server/deploy.py [hunk 1]
```diff
@@ -28,35 +31,56 @@ async def post_init_summary(source, username, added_apps, updated_apps, removed_
...
 class Deployer:
-    async def from_push_event(self, push_event):
+    def __init__(self, pusher: str, commit_message: str, current_app_definitions: AppDefinitions, previous_app_definitions: AppDefinitions):
+        self.pusher = pusher
+        self.commit_message = commit_message
+        self.current_app_definitions = current_app_definitions
+        self.previous_app_definitions = previous_app_definitions
+
+        # Max 10 helm installs at a time
+        self.semaphore = asyncio.Semaphore(10)
```
Comment (Sticky-Bits) [inline gitops_server/deploy.py:68]: :+1:

File: gitops_server/utils.py [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
     exit_code = 0
-    try:
-        output = subprocess.check_output(
-            command,
-            shell=True,
-            stderr=subprocess.STDOUT
+    logger.info(f'Running "{command}".')
+    proc = await asyncio.create_subprocess_shell(
+        command,
+        stdout=asyncio.subprocess.PIPE,
```
Comment (Sticky-Bits) [inline gitops_server/utils.py:19]: Nice. I didn't know about `asyncio.subprocess`.

File: common/utils.py [hunk 3]
```diff
@@ -1,25 +1,21 @@
 import os
 import yaml
+from typing import Dict
```
Comment (Sticky-Bits) [inline common/utils.py:3]: Why do we do this instead of just using the builtin `dict` for type hinting, as we do with `str` in other places?

**Review by Sticky-Bits — APPROVED** [review]
Nice! Good cleanup too.

Side note: I think gitops has a few too many loose util functions that could probably live as class methods, but not relevant to this PR.

File: common/utils.py (hunk 3 above)
Comment (uptickmetachu) [reply common/utils.py:3]: For python 3.8 and below, we have to use Dict to indicate the type hint for dictionaries. Python 3.9+ we will be able to use dict, tuple etc. 

Would love a python 3.9 swap in the near future.

Comment (jarekwg) [reply common/utils.py:3]: Go right ahead -- I'm not going to bother with a pbi for that one, as i don't think we'll have any backwards compatibility issues; one of us should just bump it when they wanna.

**Review by jarekwg — APPROVED** [review]
Exciting!
Only thing I'm not sure about is allowing cluster & account to be optional. I roughly recall intentionally making those required.

**Comment by uptickmetachu** [comment]
If their required we should make os.environ[''] instead of .get( "").

Was fixing type annotations with mypy. But yeah I reckon make it always there otherwise throw error loudly. I'll do that in the commit I bump the version.

**Pull request merged**